ibis/backends/tests/tpc/ds/test_queries.py

```diff
@@ -3213,6 +3213,169 @@ def test_63(item, store_sales, date_dim, store):
     )
 
 
+@tpc_test("ds")
+def test_64(
+    catalog_sales,
+    catalog_returns,
+    store_sales,
+    store_returns,
+    date_dim,
+    store,
+    customer,
+    customer_demographics,
+    promotion,
+    household_demographics,
+    customer_address,
+    income_band,
+    item,
+):
+    cs_ui = (
+        catalog_sales.join(
+            catalog_returns,
+            [
+                _.cs_item_sk == catalog_returns.cr_item_sk,
+                _.cs_order_number == catalog_returns.cr_order_number,
+            ],
+        )
+        .group_by(_.cs_item_sk)
+        .having(
+            _.cs_ext_list_price.sum()
+            > (
+                2
+                * (_.cr_refunded_cash + _.cr_reversed_charge + _.cr_store_credit).sum()
+            )
+        )
+        .agg(
+            sale=_.cs_ext_list_price.sum(),
+            refund=(
+                _.cr_refunded_cash + _.cr_reversed_charge + _.cr_store_credit
+            ).sum(),
+        )
+    )
+
+    d1 = date_dim.mutate(syear=_.d_year)
+    d2 = date_dim.view().mutate(fsyear=_.d_year)
+    d3 = date_dim.view().mutate(s2year=_.d_year)
+
+    cd1 = customer_demographics
+    cd2 = customer_demographics.view()
+
+    hd1 = household_demographics
+    hd2 = household_demographics.view()
+
+    ad1 = customer_address
+    ad2 = customer_address.view()
+
+    ib1 = income_band
+    ib2 = income_band.view()
+
+    cross_sales = (
+        store_sales.join(
+            store_returns,
+            [("ss_item_sk", "sr_item_sk"), ("ss_ticket_number", "sr_ticket_number")],
+        )
+        .join(cs_ui, _.ss_item_sk == cs_ui.cs_item_sk)
+        .join(customer, _.ss_customer_sk == customer.c_customer_sk)
+        .join(d1, _.ss_sold_date_sk == d1.d_date_sk)
+        .join(d2, _.c_first_sales_date_sk == d2.d_date_sk)
+        .join(d3, _.c_first_shipto_date_sk == d3.d_date_sk)
+        .join(store, _.ss_store_sk == store.s_store_sk)
+        .join(cd1, _.ss_cdemo_sk == cd1.cd_demo_sk)
+        .join(
+            cd2,
+            [
+                _.c_current_cdemo_sk == cd2.cd_demo_sk,
+                _.cd_marital_status != cd2.cd_marital_status,
+            ],
+        )
```

Comment:
For some reason, commenting out this join allows the query to finish.

One thing that might be in the mix here that unfortunately still isn't possible to disable is the magic variable discovery done by the DuckDB client.

I can't see any variables that would overlap here, especially in the method that's invoking `self.con.sql` (`_to_duckdb_relation`), but it does seem like that's the only thing feature-wise in the various backends' Python clients that differentiates DuckDB.

At the very least we need the ability to disable this behavior.

Reply:
I'm just deeply confused why it only happens inside `pytest`

Reply:
Yes, that doesn't make much sense.